test/envtest/deploy_resources.go

```diff
@@ -34,6 +36,33 @@ func deployKonnectAPIAuthConfiguration(
 	}
 	require.NoError(t, cl.Create(ctx, apiAuth))
 	t.Logf("deployed %s KonnectAPIAuthConfiguration resource", client.ObjectKeyFromObject(apiAuth))
+
+	apiAuth.Status.Conditions = []metav1.Condition{
+		{
+			Type:               conditions.KonnectEntityAPIAuthConfigurationValidConditionType,
+			Status:             metav1.ConditionTrue,
+			Reason:             conditions.KonnectEntityAPIAuthConfigurationReasonValid,
+			ObservedGeneration: apiAuth.GetGeneration(),
+			LastTransitionTime: metav1.Now(),
+		},
+	}
+	require.NoError(t, cl.Status().Update(ctx, apiAuth))
+	return apiAuth
+}
+
+// deployKonnectAPIAuthConfigurationWithID deploys a KonnectAPIAuthConfiguration resource and returns the resource.
+// The Status ID and Programmed condition are set on the CP using status Update() call.
+// It can be useful where the reconciler for KonnectAPIAuthConfiguration is not started
+// and hence the status has to be filled manually.
+func deployKonnectAPIAuthConfigurationWithID(
+	t *testing.T,
+	ctx context.Context,
+	cl client.Client,
+) *konnectv1alpha1.KonnectAPIAuthConfiguration {
+	t.Helper()
+
+	apiAuth := deployKonnectAPIAuthConfiguration(t, ctx, cl)
+
 	apiAuth.Status.Conditions = []metav1.Condition{
```

Comment:
It's already done in `deployKonnectAPIAuthConfiguration`, isn't it? The comment also says about populating the Konnect status, but I cannot see it happening.

Reply:
Good catch! Thanks. bbcffcbc44674042511ddced9fd67c2b192f072b should fix this.